lib/rubygems/config_file.rb

```diff
@@ -306,9 +306,18 @@ def rubygems_api_key
   # Sets the RubyGems.org API key to +api_key+
 
   def rubygems_api_key= api_key
+    set_api_key :rubygems_api_key, api_key
+
+    @rubygems_api_key = api_key
+  end
+
+  ##
+  # Set a specific host's API key to +api_key+
+
+  def set_api_key host, api_key
     check_credentials_permissions
 
-    config = load_file(credentials_path).merge(:rubygems_api_key => api_key)
+    config = load_file(credentials_path).merge(host => api_key)
```

Comment:
@drbrain should be prevented by https://github.com/rubygems/rubygems/pull/1485/files#diff-f1b45434a730b5cfbe4bac4f88479c7fR309